core/pom.xml

```diff
@@ -285,6 +285,9 @@
                                 <include>org/elasticsearch/cluster/routing/TestShardRouting.class</include>
                                 <include>org/elasticsearch/cluster/routing/TestShardRouting$*.class</include>
                                 <include>org/elasticsearch/index/shard/MockEngineFactoryPlugin.class</include>
+                                <!-- Shared between core and backwards compatibility tests. -->
+                                <include>org/elasticsearch/gateway/RecoveryFromGatewayIT.class</include>
```

Comment:
I don't believe its run but I'm happy to factor the shared code into another class - that is clearer.